other/giftcard/src/main/java/com/shortcuts/example/giftcard/GiftCardService.java

```
@@ -0,0 +1,172 @@
+        HttpPost httpPost = setupHttpPost(endpoint, requestHeader);
+        httpPost.setEntity(new StringEntity(objectMapper.writeValueAsString(requestBody)));
+        String httpResponseBody = sendPostRequest(httpPost);
+
+        CardServiceResponse response = objectMapper.readValue(httpResponseBody, CardServiceResponse.class);
+        return response;
+    }
+
+    private ObjectMapper configureObjectMapper(){
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        return objectMapper;
+    }
+
+    private HttpGet setupHttpGet(String endpoint, RequestHeader requestHeader){
+        HttpGet httpGet = new HttpGet(endpoint);
+        httpGet.addHeader("Authorization", String.format("JWT %s", requestHeader.getJwtToken()));
+        httpGet.addHeader("username", requestHeader.getSerialNumber());
+        return httpGet;
+    }
+
+    private HttpPost setupHttpPost(String endpoint, RequestHeader requestHeader){
+        HttpPost httpPost = new HttpPost(endpoint);
+        httpPost.addHeader(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType());
+        httpPost.addHeader("Authorization", String.format("JWT %s", requestHeader.getJwtToken()));
+        httpPost.addHeader("username", requestHeader.getSerialNumber());
+        return httpPost;
+    }
+
+    @SneakyThrows
+    private String sendGetRequest(HttpGet httpGet){
+        CloseableHttpResponse httpResponse = HttpClientBuilder.create().build().execute(httpGet);
+        return EntityUtils.toString(httpResponse.getEntity());
+    }
+
+    @SneakyThrows
+    private String sendPostRequest(HttpPost httpPost){
+        CloseableHttpResponse httpResponse = HttpClientBuilder.create().build().execute(httpPost);
+        return EntityUtils.toString(httpResponse.getEntity());
```

could you throw an exception if the status code is not 200